src/main/java/org/testng/asserts/SoftAssert.java

```diff
@@ -38,7 +39,12 @@ public void assertAll() {
           sb.append(",");
         }
         sb.append("\n\t");
-        sb.append(ae.getKey().getMessage());
+        AssertionError error = ae.getKey();
+        sb.append(error.getMessage());
```

Comment:
Value of ae map entry is never used.  Does it make sense to simplify?

Reply:
Fixed.